### DISCUSSION TITLE
Best way to handle "dirty" JSON response?

### DISCUSSION THREAD

**Discussion by PopGoesTheWza** [sindresorhus/got#1539] [Q&A] [answered]
#### What would you like to discuss?

I am querying a third party RESTful API which tends to serve "dirty" json responses (i.e. the `string`s may contain unescaped `\r`, `\n` and `\t` characters)

This typically causes `got/dist/source/as-promise/parse-body.js` to crash.

Right now I went "dirty" on the issue and added a very kludgy `parseJson(rawBody.toString().replace(/\r\n/g, '\\n').replace(/\n/g, '\\n').replace(/\t/g, '\\t'))` to `got/dist/source/as-promise/parse-body.js`

For a more practical solution I identify several options:

- Open an incident/request for change with the editor of this third party (will be lengthly with no guaranty to obtain a fix)
- Open a pull request here (provided handling "dirty" json is a valid feature)
- Reconfigure my `got` instance to only serve `text` responses and handle the "dirty" json separately (maybe `hooks.afterResponse` and `pagination.transform` are spots to handle it)

Any thoughts?

#### Checklist

- [x] I have read the documentation.

**Comment by Giotino** [comment, marked as answer]
In order to implement a custom JSON parser you can use the [`parseJson`](https://github.com/sindresorhus/got#parsejson) option

**Comment by PopGoesTheWza** [comment]
> In order to implement a custom JSON parser you can use the [`parseJson`](https://github.com/sindresorhus/got#parsejson) option

Hooray! That one option I have missed which makes me look like a fool!
